src/main/java/com/alkemy/ong/services/impl/RoleServiceImpl.java

```diff
@@ -16,11 +17,30 @@ public class RoleServiceImpl implements RoleService {
     RoleRepository roleRepository;
 
     @Override
-    public Set<Role> findRoleUser() {
+    public Set<Role> getRoleUser() {
         // Rol: ROLE_USER
+        Optional<Role> roleFound = roleRepository.findByName(GlobalConstants.ROLE_USER);
         Set<Role> roles = new LinkedHashSet<>();
-        roles.add(roleRepository.findByName(GlobalConstants.ROLE_USER).get());
+
+        if(roleFound.isPresent()){
+            roles.add(roleFound.get());
+        }else{
+            Role role = creationRoleUser();
+            roles.add(role);
+        }
 
         return roles;
     }
+
+    @Override
+    public Role creationRoleUser(){
+        Role role = new Role();
+
+        role.setName(GlobalConstants.ROLE_USER);
+        role.setDescription("This rol is for the common user.");
+
+        roleRepository.save(role);
+
+        return role;
```

Comment:
Todo perfecto, la última nomás q no estoy seguro: este role q se retorna, es el rol guardado y monitoreado por JPA? (es decir, tiene id?) pregunto porq como a esa variable no se le asigna el retorno de `roleRepository.save(role)`, y como el método no está como `@Transactional`, no me acuerdo cómo funcionaba en esa situación.

Reply:
Lo acabo de comprobar y sí tiene id. Luego de guardarse se le setea el id y el time stamp

![image](https://user-images.githubusercontent.com/50601878/176344164-1161d024-0c04-418d-9725-11dc14677e25.png)

Reply:
Buenísimo, to do perfecto